### PR TITLE
cli: Add `generate` command.

### PR DESCRIPTION
**Pull request by angrycub**
**Description**

 Using the `nomad-pack generate` command,  users will be able to generate a sample pack or registry using built-in starting artifacts.

**Reminders**

- [x] Add `CHANGELOG.md` entry

### CODE REVIEW

**Review by DerekStrickland — CHANGES_REQUESTED** [review]
I left a bit of feedback.  I also am wondering if a new pack should be runnable after generation.  Currently, I receive the following error.

```shell
! Failed To Process Pack

        Error:   template: pr-test/templates/_helpers.tpl:3: bad character U+002D '-'
        Type:    *errors.errorString
        Context: 
                 - Registry Name: <<local folder>>
                 - Pack Name: pr-test
                 - Pack Ref: <<none>>
                 - Pack Path: /Users/derekstrickland/goland/nomad-pack/pr-test
                 - Deployment Name: pr-test
```

File: internal/cli/generate_help.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+)
+
+// GenerateHelpCommand exists solely to provide top level help for the registry
+// set of subcommands.
+type GenerateHelpCommand struct {
+	*baseCommand
+}
+
+func (c *GenerateHelpCommand) Run(args []string) int {
+	c.cmdKey = "generate"
```
[the remark this answers is not in the document]
Comment (angrycub) [reply internal/cli/generate_help.go:15]: :shrug: It doesn't show up as dead code, and I haven't pulled that particular thread.  I can look into that in a separate effort.  Made issue #233 to track

File: internal/creator/templates/pack_metadata.hcl [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+pack {
+	name = "{{.PackName}}"
+	description = ""
+	url = ""
+	version = ""
+}
+## Optional dependency information. This block can be repeated.
+# dependency {
+# 	name = ""
+# }
```
Comment (angrycub) [inline internal/creator/templates/pack_metadata.hcl:14]:
```suggestion
# }

```

**Review by DerekStrickland — APPROVED** [review]
LGTM

**Pull request merged**